### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by grampassonnia**
Manage basic error handling on s3 bucket put requests. All or nothing approach.

### CODE REVIEW

File: src/components/UploadVideo/UploadVideo.tsx [hunk 1]
```diff
@@ -40,8 +40,8 @@ const UploadVideo = () => {
       const {filesArray, newFramesArray, videoId} = transformRawFrameData(rawFrameDataArray); // TODO refactor so we can access images from the dashboard (useContext?) and trigger the request to be, s3 and be again one after another.
       setFramesArray(newFramesArray);
       const urls: string[] = sendDataToBackEnd(newFramesArray, videoId); // TODO it's gonna be async
-      const isUploaded = await uploadImgToBucket(filesArray, urls); // make it return a boolean if responses % is high enough
-      const analytics = getAnalytics(videoId); // TODO pass analytics to helper functions and then to dashboard
+      const isUploaded = await uploadImgToBucket(filesArray, urls);
+      isUploaded && getAnalytics(videoId); // TODO pass analytics to helper functions and then to dashboard atm we're logging a string
       setAlertMessage(uploadSuccessful);
```
Comment (JackBlakeston) [inline src/components/UploadVideo/UploadVideo.tsx:45]: getAnalytics gets the data of the video analysis? Maybe call it Analysis, Analytics made me think that you were getting Analytics from the speed of the upload or something

File: src/services/s3Service.ts [hunk 2]
```diff
@@ -4,20 +4,15 @@ const batchDivider = (arr: Array<File> | Array<string>, size: number): (Array<Fi
...
-    payload.push(batch);
+    const { type } = fileArr[i];
+    const url = urlArr[i];
+    payload.push(fetch(url, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': type,
+      },
+      body: fileArr[i],
+    }));
```
Comment (JackBlakeston) [inline src/services/s3Service.ts:15]: you'll have to tell me how this works, looks promising :)

Comment (JackBlakeston) [reply src/services/s3Service.ts:15]: oh i get it after reading the next part, this is really good and clever

File: src/services/s3Service.ts [hunk 3]
```diff
@@ -27,7 +22,19 @@ export const uploadImgToBucket = async (files: File[], urls: string[], accuracy
...
+      try {
+        Promise.all(payload);
+        return result;
+      } catch (error) {
+        console.error(error);
+        result = false;
+      }
+    }
+    return result;
+  }).every(batch => batch === true);
```
Comment (JackBlakeston) [inline src/services/s3Service.ts:37]: this makes a lot of sense and looks really good

**Review by JackBlakeston — APPROVED** [review]

File: src/components/UploadVideo/UploadVideo.tsx (hunk 1 above)
Comment (grampassonnia) [reply src/components/UploadVideo/UploadVideo.tsx:45]: ahahah alright, I'll change that

**Pull request merged**